File: app/src/main/java/com/arjanvlek/oxygenupdater/updateinformation/ServerMessage.java
package com.arjanvlek.oxygenupdater.updateinformation;

import android.content.Context;

import androidx.core.content.ContextCompat;

import com.arjanvlek.oxygenupdater.R;
import com.arjanvlek.oxygenupdater.internal.i18n.Locale;
import com.fasterxml.jackson.annotation.JsonProperty;

public class ServerMessage implements Banner {
	private long id;
	private String englishMessage;
	private String dutchMessage;
	private Long deviceId;
	private Long updateMethodId;
	private ServerMessagePriority priority;

	@Override
	public String getBannerText(Context ignored) {
		return Locale.getLocale() == Locale.NL ? getDutchMessage() : getEnglishMessage();
	}

	@Override
	public int getColor(Context context) {
		switch (getPriority()) {
			case LOW:
				return ContextCompat.getColor(context, R.color.colorPositive);
			case MEDIUM:
				return ContextCompat.getColor(context, R.color.colorWarn);
			case HIGH:
				return ContextCompat.getColor(context, R.color.colorPrimary);
			default:
				return ContextCompat.getColor(context, R.color.colorPositive);
		}
	}

	public long getId() {
		return id;
	}

	public void setId(long id) {
		this.id = id;
	}

	public String getEnglishMessage() {
		return englishMessage;
	}

	@JsonProperty("english_message")
	public void setEnglishMessage(String englishMessage) {
		this.englishMessage = englishMessage;
	}

	public String getDutchMessage() {
		return dutchMessage;
	}

	@JsonProperty("dutch_message")
	public void setDutchMessage(String dutchMessage) {
		this.dutchMessage = dutchMessage;
	}

	public long getDeviceId() {
		return deviceId;
	}

	@JsonProperty("device_id")
	public void setDeviceId(Long deviceId) {
		this.deviceId = deviceId;
	}

	public long getUpdateMethodId() {
		return updateMethodId;
	}

	@JsonProperty("update_method_id")
	public void setUpdateMethodId(Long updateMethodId) {
		this.updateMethodId = updateMethodId;
	}

	public ServerMessagePriority getPriority() {
		return priority;
	}

	public void setPriority(ServerMessagePriority priority) {
		this.priority = priority;
	}

	@SuppressWarnings("WeakerAccess")
	public enum ServerMessagePriority {
		LOW,
		MEDIUM,
		HIGH
	}

}

File: app/src/main/java/com/arjanvlek/oxygenupdater/updateinformation/ServerMessage.kt
package com.arjanvlek.oxygenupdater.updateinformation

import android.content.Context
import androidx.core.content.ContextCompat
import com.arjanvlek.oxygenupdater.R
import com.arjanvlek.oxygenupdater.internal.i18n.Locale
import com.fasterxml.jackson.annotation.JsonProperty

data class ServerMessage(
    var id: Long = 0,

    @set:JsonProperty("english_message")
    var englishMessage: String? = null,

    @set:JsonProperty("dutch_message")
    var dutchMessage: String? = null,

    @set:JsonProperty("device_id")
    var deviceId: Long? = null,

    @set:JsonProperty("update_method_id")
    var updateMethodId: Long? = null,

    var priority: ServerMessagePriority? = null
) : Banner {

    override fun getBannerText(context: Context?): String? {
        return if (Locale.locale == Locale.NL) dutchMessage else englishMessage
    }

    override fun getColor(context: Context?): Int {
        return when (priority) {
            ServerMessagePriority.LOW -> ContextCompat.getColor(context!!, R.color.colorPositive)
            ServerMessagePriority.MEDIUM -> ContextCompat.getColor(context!!, R.color.colorWarn)
            ServerMessagePriority.HIGH -> ContextCompat.getColor(context!!, R.color.colorPrimary)
            else -> ContextCompat.getColor(context!!, R.color.colorPositive)
        }
    }

    enum class ServerMessagePriority {
        LOW,
        MEDIUM,
        HIGH
    }
}
